### PR TITLE
[docs] add lineinfile example

### PR DESCRIPTION
**Pull request by dminca**
##### SUMMARY

* add another example to the `lineinfile` module

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

- `lineinfile`

##### ANSIBLE VERSION

```
irrelevant
```

##### ADDITIONAL INFORMATION

_irrelevant_

```

```

### CODE REVIEW

**Comment by dminca** [comment]
Fixed commit message.

**Comment by flowerysong** [comment]
Your example is incorrect; since the regex doesn't match the line being added, it will be added every time the task runs.

```
ec2-user@mogo ~ $ cat /var/tmp/testfile 
ec2-user@mogo ~ $ ansible-playbook test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [enable IPv4 forwarding] **************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

ec2-user@mogo ~ $ ansible-playbook test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [enable IPv4 forwarding] **************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

ec2-user@mogo ~ $ ansible-playbook test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [enable IPv4 forwarding] **************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

ec2-user@mogo ~ $ cat /var/tmp/testfile 
net.ipv4.ip_forward=1
net.ipv4.ip_forward=1
net.ipv4.ip_forward=1
```

I also don't think this is a particularly useful addition; the difference between escaping in the different scalar styles is a basic YAML thing, not something specific to this module.

**Review by jborean93 — APPROVED** [review]

**Review by jborean93 — COMMENTED** [review]
Sorry didn't mean to approve this PR

File: lib/ansible/modules/files/lineinfile.py [hunk 1]
```diff
@@ -125,6 +125,13 @@
     regexp: '^SELINUX='
     line: 'SELINUX=enforcing'
 
+# use double backslash to escape between double-quotes
+- name: enable IPv4 forwarding
+  lineinfile:
+    path: /etc/sysctl.conf
+    regexp: "^#net\\.ipv4\\.ip_forward=1"
```
Comment (samdoran) [inline lib/ansible/modules/files/lineinfile.py:132]: In order for this example to not insert a line in the file every time, make the `#` optional: `^#?net\\.ipv4\\.ip_forward=1`

**Review by samdoran — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/files/lineinfile.py (hunk 1 above)
Comment (dminca) [reply lib/ansible/modules/files/lineinfile.py:132]: FIY the `/etc/sysctl.conf` has the `net.ipv4.ip_forward=1` commented out [by default](https://gist.github.com/janus57/63f88a932e41a2dd9fef#file-sysctl-conf-L28), that's the reason I've added the hashtag in the beginning. And it's idempotent, after _uncommenting_ it out it doesn't apply again, I've ran the task multiple times and it skipped.

Your test was wrong @flowerysong ...

**Comment by dminca** [comment]
Should I close this one, or it requires further investigation?   :relaxed:

**Pull request closed without merging**